src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,177 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            ShoppingCart shoppingCart = new ShoppingCart();
+            if (resultSet.next()) {
+                shoppingCart = getShoppingCartFromResultSet(resultSet);
+                return Optional.of(shoppingCart);
+            }
+            LOGGER.info("The shoppingCart " + id + " extracted");
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't extract the shoppingCart " + id, e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts ";
+        List<ShoppingCart> allShoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allShoppingCarts
+                        .add(new ShoppingCart(
+                                resultSet.getLong("cart_id"),
+                                resultSet.getLong("user_id"),
+                                getAllProducts(resultSet.getLong("cart_id"))));
+            }
+            LOGGER.info("Extracted all shoppingCarts");
+            return allShoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't extract shoppingCarts", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        deleteAllProductFromShoppingCart(id);
+        return true;
+    }
+
+    @Override
+    public boolean addProductToShoppingCart(Long productId, Long shoppingCartId) {
+        String query =
+                "INSERT INTO shopping_carts_products (cart_id, product_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, shoppingCartId);
+            statement.setLong(2, productId);
+            statement.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add the product to shoppingCart", e);
+        }
+    }
+
+    @Override
+    public boolean deleteProductFromShoppingCart(Long productId, Long shoppingCartId) {
+        String query =
+                "DELETE FROM shopping_carts_products WHERE cart_id = ? AND product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, shoppingCartId);
+            statement.setLong(2, productId);
+            statement.executeUpdate();
```

Comment:
okey, I understood your solution. You are using these methods in corresponding controllers, so you know exactly when you should delete and when add. But there can be functionality in your shop when you don't know in some place of your program how exactly cart has been changed and you don't have universal method to update it. In CRUD there are four letters;)So let's implement update too. Anyway GOOD JOB!